xdsl/transforms/canonicalization_patterns/stencil.py

```diff
@@ -11,7 +11,15 @@
 from xdsl.transforms.common_subexpression_elimination import cse
 
 
-class RedundantOperands(RewritePattern):
+class AllocUnused(RewritePattern):
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: stencil.AllocOp, rewriter: PatternRewriter) -> None:
+        if not op.field.uses:
+            rewriter.erase_op(op)
```

Comment:
This would be best done through usual DCE with an implementation of side-effect values; but I felt like moving to the application for now.